test/python/transpiler/test_mappers.py

```diff
@@ -0,0 +1,280 @@
+                           pass_manager=self.create_passmanager(coupling_map))
+        self.assertResult(result, circuit.name)
+```
+"""
+
+# pylint: disable=redefined-builtin,attribute-defined-outside-init
+
+import unittest
+import pickle
+
+from qiskit import ClassicalRegister, QuantumRegister, QuantumCircuit, BasicAer, compile
+from qiskit.transpiler import PassManager, transpile
+from qiskit.transpiler.passes import BasicSwap, LookaheadSwap, StochasticSwap
+from qiskit.mapper import CouplingMap, Layout
+
+from qiskit.test import QiskitTestCase
+
+
+class CommonUtilitiesMixin:
+    """Utilities for meta testing.
+
+    Subclasses should redefine the ``pass_class`` argument, with a Swap Mapper
+    class.
+
+    Note: This class assumes that the subclass is also inheriting from
+    ``QiskitTestCase``, and it uses ``QiskitTestCase`` methods directly.
+    """
+
+    regenerate_expected = False
+    seed = 42
+    additional_args = {}
+    pass_class = None
+
+    def create_passmanager(self, coupling_map, initial_layout=None):
+        """Returns a PassManager using self.pass_class(coupling_map, initial_layout)"""
+        passmanager = PassManager(
+            self.pass_class(CouplingMap(coupling_map),  # pylint: disable=not-callable
+                            **self.additional_args))
+        if initial_layout:
+            passmanager.property_set['layout'] = Layout(initial_layout)
+        return passmanager
+
+    def create_backend(self):
+        """Returns a Backend."""
+        return BasicAer.get_backend('qasm_simulator')
+
+    def generate_expected(self, transpiled_result, filename):
+        """Generates the expected result into a file.
+
+        Checks if transpiled_result matches self.counts by running in a backend
+        (self.create_backend()). That's saved in a pickle in filename.
+
+        Args:
+            transpiled_result (DAGCircuit): The DAGCircuit to compile and run.
+            filename (string): Where the pickle is saved.
+        """
+        sim_backend = self.create_backend()
+        qobj = compile(transpiled_result, sim_backend, seed=self.seed, shots=self.shots)
+        job = sim_backend.run(qobj)
+        self.assertDictAlmostEqual(self.counts, job.result().get_counts(), delta=self.delta)
+
+        with open(filename, "wb") as output_file:
+            pickle.dump(transpiled_result, output_file)
+
+    def assertResult(self, result, testname):
+        """Fetches the pickle in testname file and compares it with result."""
+        filename = self._get_resource_path('pickles/%s_%s.pickle' % (type(self).__name__, testname))
+
+        if self.regenerate_expected:
+            # Run result in backend to test that is valid.
+            self.generate_expected(result, filename)
+
+        with open(filename, "rb") as input_file:
+            expected = pickle.load(input_file)
+
+        self.assertEqual(result, expected)
+
+
+class CommonTestCases(CommonUtilitiesMixin):
```

Comment:
I’d rename this class to make it explicit that it contains the common test cases for swap mappers: `SwapperCommonTestCases` or `SwapperCommonTests` or something similar.

Reply:
fixed in bc5cdbbc